admin/app/view_models/workarea/admin/reports/timeline_view_model.rb

```diff
@@ -4,37 +4,74 @@ module Reports
       class TimelineViewModel < ApplicationViewModel
         include GroupByTime
 
+        class Event
```

Comment:
Hey I probably didn't communicate too well, so I started off some cleanup on this to hopefully clarify. I haven't tested this code! Happy to talk through the notes.

```ruby
class Event
  def self.build(releases, custom_events)
    # this will automatically use a new Array as the default value for each key
    result =  Hash.new { |h, k| h[k] = [] } 

    # Handling these separately will allow handling the differences pretty simply
    releases.each do |release|
      result[release.published_at.to_date] << new(ReleaseViewModel.wrap(release), type: 'release')
    end

    custom_events.each do |event|
      result[event.occured_at.to_date] << new(event, type: 'custom_event')
    end

    # Sorting and convert back to hash so you sort by the key
    result.sort.to_h
  end

  delegate_missing_to :@model
  attr_reader :model, :type

  def initialize(model, type:)
    @model = model
    @type = type
  end
end
```